### DISCUSSION TITLE
Avoid forming small grains in 2D

### DISCUSSION THREAD

**Discussion by zhangzxxl** [neperfepx/neper#950] [Q&A]
Hello everyone,
When I use the grain growth algorithm to generate the grain structure, there will always be small grain characteristics, which causes difficulties in grid mesh, please tell me how to avoid.
The following is the command I use:
neper -T -n from_morpho -morpho "gg(0.070)" -dim 2 -domain "square(1,3)"   -reg 1 -mloop 100 -ori random -oriformat plain -oridescriptor e -o RollingRVE
![RollingRVE_Geo](https://github.com/user-attachments/assets/a837019e-89fb-407a-8944-8d02e692a693)

**Comment by rquey** [comment]
It looks like the image is cropped and what you are showing are not actual cells.

Anyway, if the objective is to avoid small cells, you can specify a `from` value in the `diameq distribution`. Since `gg(0.070)` is equivalent to `diameq:lognormal(0.07,.0294),1-circularity:lognormal(0.100,0.03)`, you can use `diameq:lognormal(0.07,.0294,from=0.04),1-circularity:lognormal(0.100,0.03)`:

```
neper -T -n from_morpho -morpho "diameq:lognormal(0.07,.0294,from=0.04),1-circularity:lognormal(0.100,0.03)" -dim 2 -domain "square(1,3)" -reg 1 -mloop 100 -ori random -oriformat plain -oridescriptor e -o RollingRVE
neper -V RollingRVE.tess -dataedgerad 0.001 -print d
```

![d](https://github.com/user-attachments/assets/51e4079a-608a-48fa-9f8c-8baba6153841)

**Reply by zhangzxxl** [reply]
Dear Romain, 

Thank you for your reply！
Visualization I used this command :
neper -V RollingRVE.tess -dataedgerad 0.0002 -imagesize 4000:2000 -print RollingRVE_Geo. 
This led to the image is cropped.

Thank you again for your guidance.  For  "diameq:lognormal(0.07,.0294,from=0.04)", the term "from=0.04" has played a significant role.
